runtime/defaults/src/main/resources/application.properties

```diff
@@ -145,6 +145,8 @@ polaris.rate-limiter.token-bucket.requests-per-second=9999
 polaris.rate-limiter.token-bucket.window=PT10S
 
 polaris.active-roles-provider.type=default
+# Per-realm overrides:
+# polaris.realm1.active-roles-provider.type=custom
```

Comment:
How does `polaris.realm1.active-roles-provider.type` populate into a `QuarkusAuthenticationConfiguration` bean given that the prefix is `polaris.authentication`?
```
@ConfigMapping(prefix = "polaris.authentication")
```

Can we add a test for this?

Reply:
The mapping is handled by Quarkus per `@WithUnnamedKey(DEFAULT_REALM_KEY)` in `QuarkusAuthenticationConfiguration`

Reply:
The config prefix matters. It is associated with the `QuarkusAuthenticationConfiguration` config class.

A config entry like `polaris.active-roles-provider.type=ABC` will map to the `DEFAULT_REALM_KEY` in the [realms()](https://github.com/apache/polaris/blob/ab228afa4d975faabb7aaf1e8abb0804f5b9d353/runtime/service/src/main/java/org/apache/polaris/service/quarkus/auth/QuarkusAuthenticationConfiguration.java#L36) map.

A config entry like `polaris.realm1.active-roles-provider.type=ABC` will map to the `realm1` key in the same map.

Look up is handled by [AuthenticationConfiguration](https://github.com/apache/polaris/blob/e1c0a1ccdad4e4e432c026cdc3ae30dcb8112da1/service/common/src/main/java/org/apache/polaris/service/auth/AuthenticationConfiguration.java#L34)

Reply:
Is it OK to not follow the prefix `polaris.authentication` in this setting `polaris.active-roles-provider.type` ?

Reply:
Sorry , I made a typo in my examples and missed the prefix differences.

Reply:
I think the Quarkus config has the name convention to convert something like `active-roles-provider.type` to  `activeRolesProvider()`, which is fine. I'm still confused by the prefix.

Reply:
I guess the right new property names should be: `polaris.authentication.active-roles-provider.type=ABC` and `polaris.authentication.realm1.active-roles-provider.type=DEF`

Reply:
My apologies, I forgot to change the default `application.properties` file. And I also agree to follow the deprecation process, I will add that.

Regarding a test: that's mostly Quarkus configuration stuff, so there is little to test. But I will try to come up with something.

Reply:
Added some tests FYI, although the configuration test is imho a bit dummy because it tests mostly SmallRye config mappings.